scripts/tools/DeleteDeliveryScript.php

```diff
@@ -18,66 +32,75 @@
  * sudo -u www-data php index.php 'oat\taoDeliveryRdf\scripts\tools\DeleteDeliveryScript'
  *
  * Class TerminateSession
+ *
  * @package oat\taoDeliveryRdf\scripts\tools
  */
 class DeleteDeliveryScript extends ScriptAction
 {
-    /**
-     * @var common_report_Report
-     */
-    private $report;
-
-    protected function showTime()
+    protected function showTime(): bool
     {
         return true;
     }
 
-    protected function provideUsage()
+    protected function provideUsage(): array
     {
         return [
-            'prefix' => 'h',
-            'longPrefix' => 'help',
-            'description' => 'Prints a help statement'
+            'prefix'      => 'h',
+            'longPrefix'  => 'help',
+            'description' => 'Prints a help statement',
         ];
     }
 
-    protected function provideOptions()
+    protected function provideOptions(): array
     {
         return [
-            'delivery' => [
-                'prefix' => 'd',
-                'longPrefix' => 'delivery',
-                'required' => true,
-                'description' => 'A delivery ID.'
-            ]
+            'delivery'  => [
+                'prefix'      => 'd',
+                'longPrefix'  => 'delivery',
+                'required'    => true,
+                'description' => 'A delivery ID',
+            ],
+            'recursive' => [
+                'prefix'      => 'r',
+                'longPrefix'  => 'recursive',
+                'flag'        => true,
+                'description' => 'Remove all the linked resources such as Tests or Items',
+            ],
         ];
     }
 
-    protected function provideDescription()
+    protected function provideDescription(): string
     {
         return 'TAO Delivery - Delete Delivery';
     }
 
-    /**
-     * @return Report
-     * @throws \common_exception_Error
-     */
-    protected function run()
+    protected function run(): Report
     {
-        $this->report = common_report_Report::createInfo('Deleting Delivery ...');
+        $report = Report::createInfo('Deleting Delivery ...');
 
         /** @var DeliveryDeleteService $deliveryDeleteService */
-        $deliveryDeleteService = $this->getServiceLocator()->get(DeliveryDeleteService::SERVICE_ID);
+        $deliveryDeleteService = $this->getServiceLocator()->get(DeliveryDeleteService::class);
```

Comment:
what is the benefit of doing this ?

Reply:
Doing that ensures the service will be resolved both when it was set with a key as a FQN or as a SERVICE_ID.

A rule of thumb when working with the CG Service Locator/Manager. If you get a Service, use a FQN, if you set it, use a SERVICE_ID (until we get rid of them).